src/preloadScript.js

```diff
@@ -0,0 +1,69 @@
+import React, {Component, PropTypes} from 'react';
+import getDisplayName from 'react-display-name';
+
+const DEFAULT_SCRIPT_URL = 'https://static.opentok.com/v2/js/opentok.min.js';
+
+/*
+This higher-order component will load the OpenTok client thru a script tag.
+It will render its inner component only when the OpenTok client has loaded.
+In the meantime, it will render a loading element chosen by the developer.
+*/
+export default function preloadScript(InnerComponent) {
+  class PreloadScript extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        scriptLoaded: typeof OT !== 'undefined'
+      };
+      this.isPresent = false;
+    }
+
+    componentDidMount() {
+      this.isPresent = true;
+
+      if (this.state.scriptLoaded || this.state.scriptLoading) {
+        return;
+      }
+
+      this.setState({
+        scriptLoading: true
+      });
+
+      const scriptjs = require('scriptjs');
```

Comment:
Is there a reason this is required on-mount and not imported at the top of the file?